eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/ImpTest.hs

```diff
@@ -411,6 +419,15 @@ submitProposal ::
   ImpTestM era (GovActionId (EraCrypto era))
 submitProposal proposal = trySubmitProposal proposal >>= expectRightExpr
 
+submitProposals ::
+  (ShelleyEraImp era, ConwayEraTxBody era, HasCallStack) =>
+  NE.NonEmpty (ProposalProcedure era) ->
```

Comment:
It wouldn't be wrong to submit a transaction with zero proposals, so the non-empty list here seems to be a bit too restrictive.

Reply:
It would be wrong to submit a transaction that is intended to have proposals, that does not include proposals. See the name of the function `submitProposals`.

If you want to submit a transaction without proposals, then `submitTx` should be used.

Also, it is not excessive, it is actually very useful, since it allows to extract at least that one returned governance action id without call to `head` or `error`.